onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,121 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+/// @title Quorum Factory Test
+pragma solidity ^0.8.8;
+
+import {Test} from "forge-std/Test.sol";
+import {QuorumFactory} from "contracts/consensus/quorum/QuorumFactory.sol";
+import {Quorum} from "contracts/consensus/quorum/Quorum.sol";
+import {IHistory} from "contracts/history/IHistory.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+contract QuorumFactoryTest is Test {
+    QuorumFactory factory;
+    address[] validators;
+    uint256[] shares;
+    uint256 numValidator;
+
+    event QuorumCreated(Quorum quorum, IHistory history);
+
+    function setUp() public {
+        factory = new QuorumFactory();
+    }
+
+    function testNewQuorum(uint16 _numValidator, IHistory _history) public {
+        generateAddrAndShares(_numValidator);
+
+        vm.recordLogs();
+
+        Quorum quorum = factory.newQuorum(validators, shares, _history);
+
+        testNewQuorumAux(validators, shares, _history, quorum);
+    }
+
+    function generateAddrAndShares(uint256 _numValidator) internal {
+        // assume there are less than 1000 validators so the test wont run forever
+        vm.assume(_numValidator > 0 && _numValidator < 1000);
+        numValidator = _numValidator;
+
+        validators = new address[](numValidator);
+        shares = new uint256[](numValidator);
+        for (uint256 i; i < numValidator; ++i) {
+            validators[i] = vm.addr(i + 1); // assume there is no address collision
```

Comment:
We can guarantee that there will be no collision if addresses are assigned linearly. We know this will not create collisions because $i + 1 \le numValidator < 1000 < 2^{160}$.

```suggestion
            validators[i] = address(uint160(i + 1));
```